botocore/client.py

```diff
@@ -491,7 +492,16 @@ def _make_api_call(self, operation_name, api_params):
         )
 
         if http.status_code >= 300:
-            raise ClientError(parsed_response, operation_name)
+            try:
+                error_code = parsed_response["Error"]["Code"]
+                if "." in error_code:
+                    error_code = error_code.replace(".", "")
+                if not error_code[0].isalpha():
+                    error_code = "ClientError" + error_code
+                error_class = getattr(self.meta.exceptions, error_code)
+            except Exception:
```

Comment:
I am wondering if we should have this general try catch around the logic as it may cover up edge cases that we missed. I think I would prefer to remove the try catch or if not that at least log what exception triggered the except clause.